vnext/Desktop/WinHTTPWebSocketResource.cpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+#include "pch.h"
+
+#include <Unicode.h>
+#include <Utils.h>
+#include "WinHTTPWebSocketResource.h"
+
+// Windows API
+#include <comutil.h>
+
+using Microsoft::Common::Unicode::Utf8ToUtf16;
+
+using std::function;
+using std::size_t;
+using std::string;
+using std::wstring;
+
+namespace Microsoft::React
+{
+
+WinHTTPWebSocketResource::WinHTTPWebSocketResource(const string& urlString)
+  : m_openSessionFlags{ WINHTTP_FLAG_ASYNC }
+  , m_sessionHandle{ NULL }
+  , m_connectionHandle{ NULL }
+  , m_requestHandle{ NULL }
+  , m_errorStatus{ ERROR_SUCCESS }
+{
+#if 0 // WinHttpCrackUrl does not accept non-HTTP schemes 🙄
```

Comment:
>// WinHttpCrackUrl does not accept non-HTTP schemes 🙄 [](start = 6, length = 54)

I'm not sure what how this comment relates to the code in this region
Can the #if 0 and all the code in that section be deleted?

Reply:
The comment will be removed.
The `#if 0` section will remain for reference, for now.

Again, this class may be deleted in the next PR, once the WinRT variant is proved stable.